### PR TITLE
jumpapp: init at 1.0

### PR DESCRIPTION
**Pull request by matklad**
###### Motivation for this change

Packaging a useful utility for window management on X11: https://github.com/mkropat/jumpapp

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/X11/jumpapp/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    owner = "mkropat";
+    repo = "jumpapp";
+    rev = "v${version}";
+    sha256 = "11ibh51q4vcjkz9fqyw5dy9qrkqxm42hpdccas1s6h2dk9z62kfb";
+  };
+
+  makeFlags = [ "PREFIX=$(out)" ];
+  buildInputs = [ perl pandoc ];
+  postFixup = let
+    runtimePath = lib.makeSearchPath "bin" [ xdotool wmctrl xprop nettools perl ];
```
Comment (danbst) [inline pkgs/tools/X11/jumpapp/default.nix]: what about this approach?
```
  nativeBuildInputs = [ pandoc perl ];
  buildInputs = [ xdotool wmctrl xprop nettools ];
  postFixup = let
    runtimePath = lib.makeBinPath buildInputs;
  in ...
```

This way script will be launchable from `nix-shell`, for, e.g., script development purposes.

File: pkgs/tools/X11/jumpapp/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  ''
+    sed -i "2 i export PATH=${runtimePath}:\$PATH" $out/bin/jumpapp
+    sed -i "2 i export PATH=${perl}/bin:\$PATH" $out/bin/jumpappify-desktop-entry
+  '';
+
+  meta = {
+    homepage = https://github.com/mkropat/jumpapp;
+    description = "A run-or-raise application switcher for any X11 desktop";
+    license = stdenv.lib.licenses.mit;
+    maintainers = [ ];
```
Comment (danbst) [inline pkgs/tools/X11/jumpapp/default.nix]: would you kind sir add yourself as a maintainer, please?

File: pkgs/tools/X11/jumpapp/default.nix (hunk 1 above)
Comment (matklad) [reply pkgs/tools/X11/jumpapp/default.nix]: Neat, didn't realize that `nativeBuildInputs` can be used in this way. `perl` is both run-time and build-time dependency, so I've added it to both lists

File: pkgs/tools/X11/jumpapp/default.nix (hunk 2 above)
Comment (matklad) [reply pkgs/tools/X11/jumpapp/default.nix]: done

Comment (danbst) [reply pkgs/tools/X11/jumpapp/default.nix]: thanks!

**Pull request merged**